buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/behaviorinterface.py

```diff
@@ -0,0 +1,96 @@
+from pathlib import Path
+
+import numpy as np
+from pynwb.file import NWBFile, TimeIntervals
+
+
+from neuroconv.utils.json_schema import FolderPathType
+from neuroconv.basedatainterface import BaseDataInterface
+from neuroconv.tools.nwb_helpers import get_module
+
+from scipy.io import loadmat as loadmat_scipy
+
+
+class HuzsarBehaviorInterface(BaseDataInterface):
+    """Behavior interface"""
+
+    def __init__(self, folder_path: FolderPathType):
+        super().__init__(folder_path=folder_path)
+
+    def run_conversion(self, nwbfile: NWBFile, metadata: dict, stub_test: bool = False):
+        session_path = Path(self.source_data["folder_path"])
+        session_id = session_path.stem
+
+        module_name = "Neural states"
+        module_description = "Contains behavioral data concerning classified states."
+        processing_module = get_module(nwbfile=nwbfile, name=module_name, description=module_description)
+
+        # Sleep states
+        sleep_states_file_path = session_path / f"{session_id}.SleepState.states.mat"
+        if Path(sleep_states_file_path).exists():
+            mat_file = loadmat_scipy(sleep_states_file_path, simplify_cells=True)
+
+            state_label_names = dict(WAKEstate="Awake", NREMstate="Non-REM", REMstate="REM")
+            sleep_state_dic = mat_file["SleepState"]["ints"]
+            table = TimeIntervals(name="Sleep states", description="Sleep state of the animal.")
+            table.add_column(name="label", description="Sleep state.")
+
+            data = []
+            for sleep_state in state_label_names:
+                values = sleep_state_dic[sleep_state]
+                if len(values) != 0 and isinstance(values[0], int):
+                    values = [values]
+                for start_time, stop_time in values:
+                    data.append(
+                        dict(
+                            start_time=float(start_time),
+                            stop_time=float(stop_time),
+                            label=state_label_names[sleep_state],
+                        )
+                    )
+            [table.add_row(**row) for row in sorted(data, key=lambda x: x["start_time"])]
+            processing_module.add(table)
+
+    def align_timestamps(self, aligned_timestamps: np.ndarray):
```

Comment:
You can now remove these with https://github.com/catalystneuro/neuroconv/pull/442 if you wish